### PR TITLE
fix: `checkOrigin` headers check

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

This PR ports a fix of `checkOrigin` to v4

## Testing

CI should pass

## Docs

N/A

### CODE REVIEW

**Comment by bluwy** [comment]
Before merging this, we might want to first prepare the `4-legacy` branch for changesets release, something like this change (see first 4 files): https://github.com/sveltejs/vite-plugin-svelte/compare/main...v2

That way, changesets will create the correct PR for releasing the backport, and we don't clash the `latest` tag on npm once this backport is published. We want the `latest` tag to be the current v5 major.

**Comment by ematipico** [comment]
I don't think I understand what I have to do. I thought that adding `4-legacy` to the workflow was enough.

File: packages/astro/src/core/app/middlewares.ts [hunk 1]
```diff
@@ -21,22 +21,43 @@ const FORM_CONTENT_TYPES = [
...
-				}
+		if (request.method === "GET") {
+			return next();
+		}
+		const sameOrigin =
+			(request.method === 'POST' ||
+				request.method === 'PUT' ||
+				request.method === 'PATCH' ||
+				request.method === 'DELETE') &&
+			request.headers.get('origin') === url.origin;
```
Comment (ascorbic) [inline packages/astro/src/core/app/middlewares.ts:32]: Should we allow empty origins too?

**Review by ascorbic — APPROVED** [review]

File: packages/astro/src/core/app/middlewares.ts (hunk 1 above)
Comment (ematipico) [reply packages/astro/src/core/app/middlewares.ts:32]: I thought checking the origin was the whole point of CSRP protection 😅 Regardless, this port from v5 is a security fix. We shouldn't change the logic **in v4**, but we can enhance the v5.

**Comment by bluwy** [comment]
> I don't think I understand what I have to do. I thought that adding `4-legacy` to the workflow was enough.

I updated the `4-legacy` branch to change the npm tag. We don't need it if we release the backport, and immediately release a patch for astro 5 again. But to be safe we can prevent sharing the tag by using a different tag for the backport

**Review by bluwy — APPROVED** [review]

**Pull request merged**